### PR TITLE
RDoc-2174-Unit Tests - explained initialization and updated sample comments

### PR DESCRIPTION
**Pull request by serezhub**
Especially related to ConfigureServer( )

### CODE REVIEW

**Review by arekpalinski — APPROVED** [review]

File: Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown [hunk 1]
```diff
@@ -1,15 +1,16 @@
 # Getting Started: Writing your Unit Test using TestDriver
 
-In this section we will explain how to use [RavenDB.TestDriver](https://www.nuget.org/packages/RavenDB.TestDriver/) in order to write unit tests for working with RavenDB.
+In this section, we will explain how to use [RavenDB.TestDriver](https://www.nuget.org/packages/RavenDB.TestDriver/) in order to write unit tests for working with RavenDB.
 TestDriver uses an [Embedded](../server/embedded) package with the same set of [prerequisites](../server/embedded#prerequisites) to run the Server.
```
Comment (reebhub) [inline Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown:4]: the same set of prerequisites as what?

File: Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown [hunk 2]
```diff
@@ -44,24 +45,41 @@ Pre-Initializing the IDocumentStore allows you to mutate the conventions used by
...
+{CODE test_driver_MyFirstTest@Start\RavenDBTestDriver.cs /}
 
 {PANEL/}
 
 {PANEL: ConfigureServer}
 
-The `ConfigureServer` method allows you to be more in control on your server.  
-You can use it with `TestServerOptions` to change the path to the Raven server binaries, specify data storage path, security, .NET framework, etc.
+The `ConfigureServer` method allows you to be more in control of your server.  
+You can use it with `TestServerOptions` to change the path to the Raven server binaries, specify data storage path, security, adjust .NET framework versions, etc.
```
Comment (reebhub) [inline Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown:70]: the "security," floats there in a universe of its own

File: Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown [hunk 3]
```diff
@@ -44,24 +45,41 @@ Pre-Initializing the IDocumentStore allows you to mutate the conventions used by
...
+
+`ConfigureServer` can only be set once per test run.  
+It needs to be set before `DocumentStore` is called.  
+See an [example](../start/test-driver#complete-example) below.  
+
+If it is called twice, or within the `DocumentStore` scope, you will get the following error message:
+
+> System.InvalidOperationException : Cannot configure server after it was started. Please call 'ConfigureServer' method before any 'GetDocumentStore' is called.  
+
+{NOTE/}
```
Comment (reebhub) [inline Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown:83]: get everything inside the note out of it, make it two bullets (`condigureServer`.. and If it..)
remove the ">" and make the quote blue text

File: Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown [hunk 4]
```diff
@@ -72,6 +90,10 @@ how the embedded server is going to run with just a minor [definition change](..
 
 {INFO /}
```
Comment (reebhub) [inline Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown:91]: "See the complete list of.." see it where?

File: Documentation/4.2/Samples/csharp/Raven.Documentation.Samples/Start/RavenDBTestDriver.cs [hunk 5]
```diff
@@ -49,7 +55,7 @@ public void TestDriveConfigureServer()
 
             var testServerOptions = new TestServerOptions
             {
-                FrameworkVersion = "3.1.15+", // Look for newest version on machine including 3.1.15 and up (default is set at time of server release).  
+                FrameworkVersion = "3.1.15+", // Look for newest version on machine including 3.1.15 and any newer patches, but not major new releases (default is .NET version at time of server release).
```
Comment (reebhub) [inline Documentation/4.2/Samples/csharp/Raven.Documentation.Samples/Start/RavenDBTestDriver.cs:58]: * place all the notes here above the code lines they refer to
* break the notes to shorter lines so the reader doesn't have to scroll the scree
* add line breaks between the code lines (make it three pairs, each pair a note and beneath it the code it relates to)

File: Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown [hunk 6]
```diff
@@ -72,6 +90,10 @@ how the embedded server is going to run with just a minor [definition change](..
 
 {INFO /}
 
+#### Example
+
+{CODE test_driver_ConfigureServer@Start\RavenDBTestDriver.cs /}
+
 #### .NET FrameworkVersion
```
Comment (reebhub) [inline Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown:97]: why is .NET FrameworkVersion discussed so thoroughly here, and simply by copying the very text it links to in the end?
move the "TestServerOptions" info box here, and place two bullets in it: the first with the current `TestServerOptions` explanation, and the second with just a clear reference to where `.NET FrameworkVersion` is already explained.

File: Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown [hunk 7]
```diff
@@ -83,14 +105,18 @@ all .NET versions 3.1.17 and newer patch releases (.17, .18, .19, etc...). So, a
...
 {PANEL:Complete Example}
 
+This is a full unit test using [Xunit](https://www.nuget.org/packages/xunit/).
+
+We get an IDocumentStore to our test database, deploy an index, and insert two documents into it. 
+We then wait for the indexing to complete and launch the Studio so we can verify that the documents 
+and index are deployed (we can remove this line once the test is working).
+At the end of the test, we query for TestDocument where their name contains the word 'hello', 
+and we assert that we have only one such document.
+
```
Comment (reebhub) [inline Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown:119]: a previous comment referred to a similar pre-sample text, please implement it here as well,

**Review by reebhub — CHANGES_REQUESTED** [review]

File: Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown [hunk 8]
```diff
@@ -1,15 +1,16 @@
 # Getting Started: Writing your Unit Test using TestDriver
 
-In this section we will explain how to use [RavenDB.TestDriver](https://www.nuget.org/packages/RavenDB.TestDriver/) in order to write unit tests for working with RavenDB.
+In this section, we will explain how to use [RavenDB.TestDriver](https://www.nuget.org/packages/RavenDB.TestDriver/) in order to write unit tests for working with RavenDB.
```
[the remark this answers is not in the document]
Comment (serezhub) [reply Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown]: done

File: Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown (hunk 1 above)
[the remark this answers is not in the document]
Comment (serezhub) [reply Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown]: done

File: Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown [hunk 9]
```diff
@@ -44,24 +45,41 @@ Pre-Initializing the IDocumentStore allows you to mutate the conventions used by
 {PANEL/}
 
 {PANEL:UnitTest}
-We'll be using [xunit](https://www.nuget.org/packages/xunit/) for my test framework in the below example.
-Note that the test itself is meant to show different capabilities of the test driver and is not meant to be the most efficient.
+
+We'll be using [xunit](https://www.nuget.org/packages/xunit/) for the test framework in the below example.
```
[the remark this answers is not in the document]
Comment (serezhub) [reply Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown]: done

File: Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown [hunk 10]
```diff
@@ -44,24 +45,41 @@ Pre-Initializing the IDocumentStore allows you to mutate the conventions used by
 {PANEL/}
 
 {PANEL:UnitTest}
-We'll be using [xunit](https://www.nuget.org/packages/xunit/) for my test framework in the below example.
-Note that the test itself is meant to show different capabilities of the test driver and is not meant to be the most efficient.
+
+We'll be using [xunit](https://www.nuget.org/packages/xunit/) for the test framework in the below example.  
+
+> Note that the test itself is meant to show different capabilities of the test driver and is not meant to be the most efficient.
```
[the remark this answers is not in the document]
Comment (serezhub) [reply Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown]: done

File: Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown [hunk 11]
```diff
@@ -44,24 +45,41 @@ Pre-Initializing the IDocumentStore allows you to mutate the conventions used by
...
+We'll be using [xunit](https://www.nuget.org/packages/xunit/) for the test framework in the below example.  
+
+> Note that the test itself is meant to show different capabilities of the test driver and is not meant to be the most efficient.  
+
 The example below depends on the `TestDocumentByName` index and `TestDocument` class that can be seen in the [full example](../start/test-driver#complete-example)
 
 ### Example
 
-{CODE test_driver_MyFirstTest@Start\RavenDBTestDriver.cs /}
+In the test, we get an IDocumentStore to our test database. Deploy an index and insert two documents into it.
```
[the remark this answers is not in the document]
Comment (serezhub) [reply Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown]: done

File: Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown [hunk 12]
```diff
@@ -44,24 +45,41 @@ Pre-Initializing the IDocumentStore allows you to mutate the conventions used by
...
+
+> Note that the test itself is meant to show different capabilities of the test driver and is not meant to be the most efficient.  
+
 The example below depends on the `TestDocumentByName` index and `TestDocument` class that can be seen in the [full example](../start/test-driver#complete-example)
 
 ### Example
 
-{CODE test_driver_MyFirstTest@Start\RavenDBTestDriver.cs /}
+In the test, we get an IDocumentStore to our test database. Deploy an index and insert two documents into it. 
+We then wait for the indexing to complete and launch the Studio so we can verify that the documents
```
[the remark this answers is not in the document]
Comment (serezhub) [reply Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown]: done

File: Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown (hunk 2 above)
Comment (serezhub) [reply Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown]: done

File: Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown (hunk 3 above)
Comment (serezhub) [reply Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown:82]: done

File: Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown (hunk 6 above)
Comment (serezhub) [reply Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown]: done

File: Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown (hunk 7 above)
Comment (serezhub) [reply Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown:110]: done

File: Documentation/4.2/Samples/csharp/Raven.Documentation.Samples/Start/RavenDBTestDriver.cs (hunk 5 above)
Comment (serezhub) [reply Documentation/4.2/Samples/csharp/Raven.Documentation.Samples/Start/RavenDBTestDriver.cs:58]: done.

File: Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown [hunk 13]
```diff
@@ -1,13 +1,12 @@
 # Getting Started: Writing your Unit Test using TestDriver
 
-In this section, we will explain how to use [RavenDB.TestDriver](https://www.nuget.org/packages/RavenDB.TestDriver/) in order to write unit tests for working with RavenDB.
-TestDriver uses an [Embedded](../server/embedded) package with the same set of [prerequisites](../server/embedded#prerequisites) to run the Server.
+In this section, we explain how to use [RavenDB.TestDriver](https://www.nuget.org/packages/RavenDB.TestDriver/) to write RavenDB unit tests.
```
Comment (reebhub) [inline Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown:3]: * make it a bullet
* add line break at the end

File: Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown (hunk 3 above)
Comment (reebhub) [reply Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown:82]: indent the note, it's not with its bullet

File: Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown [hunk 14]
```diff
@@ -1,13 +1,12 @@
 # Getting Started: Writing your Unit Test using TestDriver
 
-In this section, we will explain how to use [RavenDB.TestDriver](https://www.nuget.org/packages/RavenDB.TestDriver/) in order to write unit tests for working with RavenDB.
-TestDriver uses an [Embedded](../server/embedded) package with the same set of [prerequisites](../server/embedded#prerequisites) to run the Server.
+In this section, we explain how to use [RavenDB.TestDriver](https://www.nuget.org/packages/RavenDB.TestDriver/) to write RavenDB unit tests.
+TestDriver uses an [Embedded](../server/embedded) package with the same set of [prerequisites](../server/embedded#prerequisites) as embedded servers to run the Server.
```
[the remark this answers is not in the document]
Comment (serezhub) [reply Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown:4]: done

File: Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown (hunk 13 above)
Comment (serezhub) [reply Documentation/4.2/Raven.Documentation.Pages/start/test-driver.dotnet.markdown:3]: done

**Pull request merged**